### PR TITLE
Make allowances truly transitive

### PR DESCRIPTION
**Pull request by whatyouhide**
Discussed with @josevalim.

### CODE REVIEW

File: lib/nimble_ownership.ex [hunk 1]
```diff
@@ -117,15 +117,15 @@ defmodule NimbleOwnership do
...
 
-  Use this function when `owner_pid` is allowed access to `key`, and you want
+  Use this function when `pid_with_access` is allowed access to `key`, and you want
   to also allow `pid_to_allow` to use `key`.
 
   This function return an error in the following cases:
 
     * When `pid_to_allow` is already allowed to use `key` via **another owner PID**
-      that is not `owner_pid`. In this case, the `:reason` field of the returned
+      that is not `pid_with_access`. In this case, the `:reason` field of the returned
```
Comment (josevalim) [inline lib/nimble_ownership.ex:128]:
```suggestion
      that is not the owner of `pid_with_access`. In this case, the `:reason` field of the returned
```

File: lib/nimble_ownership.ex [hunk 2]
```diff
@@ -142,6 +142,16 @@ defmodule NimbleOwnership do
...
 
+  ### Transitive Allowances
+
+  Allowances are **transitive**. What this means is that the allowance of `pid_to_allow`
+  on `key` does not depend on `pid_with_access` if `pid_with_access` is not the original
+  owner of `key`. That is, if `pid_with_access` is just a PID that is allowed to access
+  `key` through some other owner PID, then `pid_to_allow`'s allowance is not tied to
+  `pid_with_access` after the allowance is made. In other words, `pid_with_access` is only
+  used to allow access to `key`. For example, if `pid_with_access` allows `pid_to_allow`
+  but then goes down, `pid_to_allow` is still allowed to access `key`.
```
Comment (josevalim) [inline lib/nimble_ownership.ex:153]:
```suggestion
  ### Transitive Allowances

  Allowances are **transitive**. `pid_with_access` allows `pid_to_allow`, it is equivalent
  to the owner of `pid_with_access` allowing `pid_to_allow`, effectively tying `pid_to_allow`
  with the owner. If `pid_with_access` terminates, `pid_to_allow`will still have access to the
  key, until the `owner_pid` itself terminates or removes the allowance.
```

Comment (josevalim) [reply lib/nimble_ownership.ex:153]: I think the previous version was focusing too much on what doesn't happen. I wrote what happens. :)

**Review by josevalim — APPROVED** [review]

File: lib/nimble_ownership.ex (hunk 2 above)
Comment (whatyouhide) [reply lib/nimble_ownership.ex:153]: Nice!

**Pull request merged**